node/task.ts

```diff
@@ -725,36 +728,104 @@ export function resolve(...pathSegments: any[]): string {
-        return shell.ls(paths);
-    }
+export function ls(paths: string[], recursive: boolean = false, allFiles: boolean = false): string[] {
+    let filePaths: string[] = [];
+    paths.forEach(path => {
+        let commandString = '';
+        if (getPlatform() == Platform.Windows) {
+            commandString = 'dir ' + path + ' /b';
+            if (recursive) {
+                commandString += ' /s';
+            }
+            if (allFiles) {
+                commandString += ' /a';
+            }
+        }
+        else {
+            commandString = 'ls ' + path;
+            if (recursive) {
+                commandString += ' -R';
+            }
+            if (allFiles) {
+                commandString += ' -A';
+            }
+        }
+        child_process.execSync(commandString).toString().split(os.EOL).forEach((file: string) => {
+            if (filePaths.indexOf(file) < 0) {
+                filePaths.push(file);
+            }
+        });
+    });
+
+    return filePaths;
 }
 
 /**
  * Copies a file or folder.
  * 
  * @param     source     source path
  * @param     dest       destination path
- * @param     options    string -r, -f or -rf for recursive and force 
+ * @param     recursive  Copy recursively
+ * @param     force      Force copy
  * @param     continueOnError optional. whether to continue on error
  */
-export function cp(source: string, dest: string, options?: string, continueOnError?: boolean): void {
-    if (options) {
-        shell.cp(options, source, dest);
+export function cp(source: string, dest: string, recursive: boolean = false, force: boolean = false, continueOnError?: boolean): void {
+    try {
+        if (!fs.existsSync(source)) {
+            throw new Error(loc('LIB_OperationFailed', source + ' doesnt exist'));
+        }
+        if (getPlatform() == Platform.Windows) {
+            if (fs.statSync(source).isDirectory()) {
+                mkdirP(dest);
+                let commandString = 'robocopy ' + source + ' ' + dest;
+                
+                if (recursive) {
+                    commandString += ' /e';
+                }
+                if (force) {
+                    commandString += ' /is /it';
+                }
+
+                child_process.execSync(commandString);
+            }
+            else {
+                // Copy individual file over
+                if (fs.existsSync(dest)) {
+                    if (force) {
+                        fs.unlinkSync(dest);
+                    }
+                    else {
+                        // If file exists and we're not overwriting, just return. 
+                        return;
+                    }
+                }
+                let commandString = 'echo F | xcopy ' + source + ' ' + dest;
+                child_process.execSync(commandString);
+            }
+        }
+        else {
+            let commandString = 'cp ';
+            if (recursive) {
+                commandString += '-R ';
```

Comment:
Should we be consistent with above? Add space before param.

Reply:
Sure